prime-router/src/test/kotlin/fhirengine/translation/hl7/utils/CustomFHIRFunctionsTests.kt

```diff
@@ -0,0 +1,265 @@
+package gov.cdc.prime.router.fhirengine.translation.hl7.utils
+
+import assertk.assertThat
+import assertk.assertions.isEmpty
+import assertk.assertions.isEqualTo
+import assertk.assertions.isFailure
+import assertk.assertions.isNotEmpty
+import assertk.assertions.isNotNull
+import assertk.assertions.isNull
+import assertk.assertions.isSuccess
+import org.hl7.fhir.r4.model.Base
+import org.hl7.fhir.r4.model.DateTimeType
+import org.hl7.fhir.r4.model.IntegerType
+import org.hl7.fhir.r4.model.MessageHeader
+import org.hl7.fhir.r4.model.OidType
+import org.hl7.fhir.r4.model.StringType
+import org.junit.jupiter.api.Test
+import java.util.UUID
+
+class CustomFHIRFunctionsTests {
+    @Test
+    fun `test get function name enum`() {
+        assertThat(CustomFHIRFunctions.CustomFHIRFunctionNames.get(null)).isNull()
+        assertThat(CustomFHIRFunctions.CustomFHIRFunctionNames.get("someBadName")).isNull()
+        val goodName = CustomFHIRFunctions.CustomFHIRFunctionNames.GetId.name
+        assertThat(CustomFHIRFunctions.CustomFHIRFunctionNames.get(goodName)).isNotNull()
+        val nameFormattedFromFhirPath = goodName.replaceFirstChar(Char::lowercase)
+        assertThat(CustomFHIRFunctions.CustomFHIRFunctionNames.get(nameFormattedFromFhirPath)).isNotNull()
+    }
+
+    @Test
+    fun `test resolve function name`() {
+        assertThat(CustomFHIRFunctions.resolveFunction(null)).isNull()
+        assertThat(CustomFHIRFunctions.resolveFunction("someBadName")).isNull()
+        val nameFormattedFromFhirPath = CustomFHIRFunctions.CustomFHIRFunctionNames.GetId.name
+            .replaceFirstChar(Char::lowercase)
+        assertThat(CustomFHIRFunctions.resolveFunction(nameFormattedFromFhirPath)).isNotNull()
```

Comment:
I was a little confused by this, just because I wasn't sure why this was only done for `GetId`. Would it be okay to bring this check into the `forEach` below, so that we check the lowercase version for every function? Either that, or maybe a comment before initializing `nameFormattedFromFhirPath` to explain that we're just checking it for one of the functions.

Reply:
There is no need to do it for every function. This is only testing the code that converts the first character, which is used for all the functions.  The foreach is testing that each function in the enum actually has a function it maps too.  Two different tests.